### PR TITLE
docs: Storybook 7 migrate

### PR DESCRIPTION
**Pull request by SergeyRoyt**
https://monday.monday.com/boards/3532714909/pulses/3540156731

### CODE REVIEW

File: .storybook/main.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  addons: getAddons(),
+  framework: {
+    name: "@storybook/react-webpack5",
+    options: {}
+  },
+  docs: {
+    autodocs: false
+  },
+  features: {
+    // interactionsDebugger: true // TODO storybook 7 migration: from storybook 6 - not sure it'll work here, need to test
```
Comment (SergeyRoyt) [inline .storybook/main.ts:36]: TODO

File: .storybook/manager.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// import { SidebarItem } from "vibe-storybook-components";
+// import "vibe-storybook-components/index.css";
+import theme from "./theme";
+
+window.STORYBOOK_GA_ID = "UA-308574295";
+window.STORYBOOK_REACT_GA_OPTIONS = {};
+
+addons.setConfig({
+  theme: theme
+  // TODO storybook 7 migration: sidebar tags not working (at least without additional efforts) in storybook 7
```
Comment (SergeyRoyt) [inline .storybook/manager.jsx:12]: TODO

File: plop/general/component-stories-mdx.txt [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { createStoryMetaSettingsDecorator } from "../../../storybook";
 import { createComponentTemplate } from "vibe-storybook-components";
 import styles from "./{{properCase componentName}}.stories.module.scss";
 
+// TODO storybook 7 migration: change plop template for Storybook 7 MDX2 format
```
Comment (SergeyRoyt) [inline plop/general/component-stories-mdx.txt:7]: TODO

File: src/components/Accordion/Accordion/__tests__/accordion.interactions.ts [hunk 4]
```diff
@@ -28,6 +28,7 @@ async function openAndCheckAccordionItem(canvas: Screen, title: string) {
   const before = getOpenedAccordionItem(canvas);
   const elHeading = getAccordionHeadingByText(canvas, title);
   userEvent.click(elHeading);
+  await delay(CHANGES_DELAY);
```
Comment (shlomitc) [inline src/components/Accordion/Accordion/__tests__/accordion.interactions.ts:31]: we prefer actively waiting for something to happen (polling) and not "sleeping".

File: package.json [hunk 5]
```diff
@@ -72,8 +72,8 @@
     "build:esm": "rollup -c",
     "build:esm:mock-classnames": "mock_classnames=on rollup -c",
     "build:react-icons": "svg2react-icon --typescript --keep-colors node_modules/monday-ui-style/src/Icons src/components/Icon/Icons",
-    "storybook": "storybook=true start-storybook -p 7008",
-    "build-storybook": "npm run build:storybook-exports && storybook=true build-storybook",
+    "storybook": "storybook=true storybook dev -p 7008 --no-open",
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:75]: That means that running storybook won't open it in a new window each time - convenient for me, but tell me wdyt about it

File: package.json [hunk 6]
```diff
@@ -140,24 +143,29 @@
...
+    "@storybook/addon-links": "^7.5.3",
+    "@storybook/addon-storyshots": "^7.5.3",
+    "@storybook/addon-styling-webpack": "^0.0.5",
+    "@storybook/addon-themes": "^7.5.3",
+    "@storybook/addon-toolbars": "^7.5.3",
+    "@storybook/addons": "^7.5.3",
+    "@storybook/blocks": "^7.5.3",
+    "@storybook/jest": "^0.2.3",
+    "@storybook/manager-api": "^7.5.3",
+    "@storybook/mdx1-csf": "^1.0.0",
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:161]: Converter which was used to convert from MDX1 to MDX2 - removing, thanks

File: package.json [hunk 7]
```diff
@@ -140,24 +143,29 @@
...
+    "@storybook/addon-essentials": "^7.5.3",
+    "@storybook/addon-interactions": "^7.5.3",
+    "@storybook/addon-links": "^7.5.3",
+    "@storybook/addon-storyshots": "^7.5.3",
+    "@storybook/addon-styling-webpack": "^0.0.5",
+    "@storybook/addon-themes": "^7.5.3",
+    "@storybook/addon-toolbars": "^7.5.3",
+    "@storybook/addons": "^7.5.3",
+    "@storybook/blocks": "^7.5.3",
+    "@storybook/jest": "^0.2.3",
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:159]: It's being used, yeah

File: package.json [hunk 8]
```diff
@@ -140,24 +143,29 @@
...
-    "@storybook/jest": "^0.0.10",
-    "@storybook/manager-webpack5": "^6.5.10",
+    "@storybook/addon-a11y": "^7.5.3",
+    "@storybook/addon-actions": "^7.5.3",
+    "@storybook/addon-controls": "^7.5.3",
+    "@storybook/addon-docs": "^7.5.3",
+    "@storybook/addon-essentials": "^7.5.3",
+    "@storybook/addon-interactions": "^7.5.3",
+    "@storybook/addon-links": "^7.5.3",
+    "@storybook/addon-storyshots": "^7.5.3",
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:153]: It's being used, yeah

File: package.json [hunk 9]
```diff
@@ -140,24 +143,29 @@
...
-    "@storybook/addon-toolbars": "^6.5.9",
-    "@storybook/addons": "^6.5.10",
-    "@storybook/builder-webpack5": "^6.5.10",
-    "@storybook/jest": "^0.0.10",
-    "@storybook/manager-webpack5": "^6.5.10",
+    "@storybook/addon-a11y": "^7.5.3",
+    "@storybook/addon-actions": "^7.5.3",
+    "@storybook/addon-controls": "^7.5.3",
+    "@storybook/addon-docs": "^7.5.3",
+    "@storybook/addon-essentials": "^7.5.3",
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:150]: 100%

File: src/components/Accordion/Accordion/__tests__/accordion.interactions.ts (hunk 4 above)
Comment (SergeyRoyt) [reply src/components/Accordion/Accordion/__tests__/accordion.interactions.ts:31]: I don't mind to try it out, just tell me how it should look, cause I found only this pattern in our interactions tests

File: plop/general/component-stories-scss.txt [hunk 10]
```diff
@@ -1,3 +1,3 @@
-.{{camelCase componentName}} {
+.{{camelCase componentName}}Story {
```
[the remark this answers is not in the document]
Comment (shlomitc) [reply plop/general/component-stories-scss.txt:1]: It’s creating code snippets which aren’t reflected in the documentation

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (talkor) [reply package.json:75]: I think it's useful that it opens, it would be also easier for external contributors

File: package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (talkor) [reply package.json:153]: Isn't it deprecated? We do we use it?

File: plop/component/index.js [hunk 11]
```diff
@@ -27,9 +27,25 @@ module.exports = plop => {
       },
       {
         type: "add",
-        path: "src/components/{{properCase componentName}}/__stories__/{{properCase componentName}}.stories.mdx",
+        path: "src/components/{{properCase componentName}}/__stories__/{{properCase componentName}}.mdx",
```
Comment (talkor) [inline plop/component/index.js:30]: Why do some files have the `.stories.` and some don't?

File: plop/general/component-stories-helpers-js.txt [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import { Link, Tip } from "vibe-storybook-components";
+
+export const TipOtherComponent = () => (
```
Comment (talkor) [inline plop/general/component-stories-helpers-js.txt:3]: Why is this a "helper" and not a component in `storybook-vibe-components`?

File: src/components/Accordion/Accordion/__tests__/accordion.interactions.ts (hunk 4 above)
Comment (talkor) [reply src/components/Accordion/Accordion/__tests__/accordion.interactions.ts:31]: waitFor should work?

File: src/storybook/stand-alone-documentaion/typography/typography.mdx [hunk 13]
```diff
@@ -1,17 +1,16 @@
...
-import { TEXT, HEADING, EDITABLE_HEADING } from "../../components/related-components/component-description-map";
+import { EDITABLE_HEADING, HEADING, TEXT } from "../../components/related-components/component-description-map";
 import { typographyExample1, typographyExample2, typographyExample3, typographyExample4 } from "./assets";
 import ExamplesTable from "./examples-table/examples-table";
+import { TipHowToUseFonts } from "./typograpghy.stories.helpers";
 import styles from "./typography.module.scss";
 
 <Meta title="Foundations/Typography" />
 
-<!--- Component documentation -->
```
Comment (talkor) [inline src/storybook/stand-alone-documentaion/typography/typography.mdx:13]: Thank god, I hated these comments

File: src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.js [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+  title: "Hooks/useVibeMediaQuery"
+};
+
+export const Overview = {
+  render: () => {
+    const currentSize = useVibeMediaQuery();
+    return <div className={`hooks-reference-element ${styles.container}`}>{currentSize}</div>;
+  },
+
+  name: "Overview"
```
Comment (talkor) [inline src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.js:15]: Isn't it redundant? It takes the function's name

File: src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx [hunk 15]
```diff
@@ -124,11 +106,8 @@ Attaches a listener to keyboard DOM events on a specific element, firing a provi
       type="string"
       description={
         <>
-          The event to listen to. See{" "}
-          <Link
-            href="https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent#events" withoutSpacing>
```
Comment (talkor) [inline src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx:129]: Isn't it better with the `withoutSpacing` prop? Instead of margin it adds a blank space. I wonder whether is should be the default

File: src/components/Menu/MenuItem/__stories__/MenuItem.stories.js [hunk 16]
```diff
@@ -6,36 +6,74 @@ import { Activity, Bolt, Settings } from "../../../Icon/Icons";
...
 import Tooltip from "../../../Tooltip/Tooltip";
 import DialogContentContainer from "../../../DialogContentContainer/DialogContentContainer";
+import { createStoryMetaSettingsDecorator } from "../../../../storybook/functions/createStoryMetaSettingsDecorator";
 
-export const menuItemTemplate = args => (
+const metaSettings = createStoryMetaSettingsDecorator({
+  component: MenuItem,
+  enumPropNamesArray: ["tooltipPosition"],
+  iconPropNamesArray: ["icon"]
+});
```
Comment (talkor) [inline src/components/Menu/MenuItem/__stories__/MenuItem.stories.js:15]: I'm not sure we need these properties anymore, I think you can provide types in JSDoc and Storybook will use them as a list of values. I think I did it in EditableHeading/Text. Not sure what he `iconPropsNameArray` though

File: src/components/Menu/MenuItem/__stories__/MenuItem.stories.js [hunk 17]
```diff
@@ -47,11 +85,12 @@ export const menuItemOverflowTemplate = args => {
...
     </Menu>
-  );
+  ),
+  name: "Overflow"
 };
 
-export const menuItemTooltipTemplate = args => {
-  return (
+export const TooltipStory = {
+  render: () => (
```
Comment (talkor) [inline src/components/Menu/MenuItem/__stories__/MenuItem.stories.js:93]: If the name is not provided we can just return a function

File: src/components/Menu/MenuDivider/__stories__/MenuDivider.stories.js [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+import { createComponentTemplate } from "vibe-storybook-components";
+import Menu from "../../Menu/Menu";
+import MenuItem from "../../MenuItem/MenuItem";
+
+export default {
+  title: "Navigation/Menu/MenuDivider",
+  component: MenuDivider
+};
+
+const menuDividerTemplate = createComponentTemplate(MenuDivider);
```
Comment (talkor) [inline src/components/Menu/MenuDivider/__stories__/MenuDivider.stories.js:11]: Why not just write the code? We don't bind it anywhere

File: src/components/Accordion/AccordionItem/__stories__/accordionItem.stories.tmp
Comment (talkor) [inline src/components/Accordion/AccordionItem/__stories__/accordionItem.stories.tmp:1]: Any change we can be consistent and use PascalCase file all component/stories filenames?

File: src/components/Badge/__stories__/Badge.stories.js [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+import Flex from "../../Flex/Flex";
+import Avatar from "../../Avatar/Avatar";
+import { createStoryMetaSettingsDecorator } from "../../../storybook";
+import { createComponentTemplate, StoryDescription } from "vibe-storybook-components";
+
+const metaSettings = createStoryMetaSettingsDecorator({
+  component: Badge,
+  enumPropNamesArray: ["anchor", "alignment", "type"],
+  iconPropNamesArray: [],
+  actionPropsArray: [],
```
Comment (talkor) [inline src/components/Badge/__stories__/Badge.stories.js:15]: Same comment about this decorator, I think 100% of our actions start with `on*`, I'm pretty sure we can set it globally to identify these props as actions and log them in the addon

File: src/components/EditableHeading/__tests__/EditableHeading-interactions.tsx [hunk 20]
```diff
@@ -26,8 +28,10 @@ function getInput(canvas: Canvas) {
 }
 
 async function changeModes(canvas: Canvas) {
+  await delay(CHANGES_DELAY); // needed the tests would run correctly on page refresh
```
Comment (talkor) [inline src/components/EditableHeading/__tests__/EditableHeading-interactions.tsx:31]: What happens when the page refreshes?

File: src/components/ListTitle/__stories__/listTitle.stories.temp
Comment (talkor) [inline src/components/ListTitle/__stories__/listTitle.stories.temp:1]: Why temp?

File: src/components/Accordion/AccordionItem/__stories__/accordionItem.stories.tmp
Comment (SergeyRoyt) [reply src/components/Accordion/AccordionItem/__stories__/accordionItem.stories.tmp:1]: I'd rename everything but didn't want to create even more changes all at once by renaming tons of files :) 
Maybe another time

File: src/components/ListTitle/__stories__/listTitle.stories.temp
Comment (SergeyRoyt) [reply src/components/ListTitle/__stories__/listTitle.stories.temp:1]: It wasn't displayed before, probably need some work around it
This way I'm naming drafts for stories

File: package.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:75]: Maybe u r right, I'll remove it

File: package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:153]: We are using it to take the story snapshots
It's indeed seems deprecated now - they propose a different way to achieve the same functionality, I'll add a TODO to migrate it

File: plop/general/component-stories-scss.txt (hunk 10 above)
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply plop/general/component-stories-scss.txt:1]: But they might be used when u r creating an actual story
Not sure, it's debateful

File: plop/component/index.js (hunk 11 above)
Comment (SergeyRoyt) [reply plop/component/index.js:30]: `.mdx` files shouldn't have `.stories.` prefix now, that's the storybook 7 law kinda - they just won't render
Rest of the stories related files is with `.stories.` prefix

File: plop/general/component-stories-helpers-js.txt (hunk 12 above)
Comment (SergeyRoyt) [reply plop/general/component-stories-helpers-js.txt:3]: No, it's different
This is just a `Tip` which you render in a mdx file with a link to some other component
I'd render it normally in the mdx without any helpers as we did before, but starting from storybook 7, we have an issue with that

I'll give an example, just so it'd be easier. Let's say we have a tip like this
```
  <Tip title="Use color contrast tools for code and design">
    <div>
      To check color contrast in design use
      <Link href="https://www.figma.com/community/plugin/748533339900865323/Contrast" size={Link.sizes.SMALL}>
        contrast figma plugin
      </Link>
      and inspect elements accessibility tab for live code verification
    </div>
  </Tip>
```
If you'd render it in mdx it'll look like this
image
Cause this stuff is treating text around the `Link` like `p` and it's causing this

You could have solved it by wrapping all the text in `span` but I thought it's not really convenient.
Besides it's not the only use case when mdx is playing tricks with you, so that's why I found this `.stories.helpers.js` workaround

File: src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.js (hunk 14 above)
Comment (SergeyRoyt) [reply src/hooks/useVibeMediaQuery/__stories__/useVibeMediaQuery.stories.js:15]: It is mostly
It's automatic converter job
Though I guess there are cases when it's needed

File: src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx (hunk 15 above)
Comment (SergeyRoyt) [reply src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx:129]: This stuff was the build-killer
image

Turns out that new converter they use doesn't allow such syntax -

File: src/components/Menu/MenuItem/__stories__/MenuItem.stories.js (hunk 17 above)
Comment (SergeyRoyt) [reply src/components/Menu/MenuItem/__stories__/MenuItem.stories.js:93]: Yeah, but again automatic conversion, which I was too lazy too rewrite completely

File: src/components/Menu/MenuDivider/__stories__/MenuDivider.stories.js (hunk 18 above)
Comment (SergeyRoyt) [reply src/components/Menu/MenuDivider/__stories__/MenuDivider.stories.js:11]: We bind it in overview example and provide args for it

File: src/components/Badge/__stories__/Badge.stories.js (hunk 19 above)
Comment (SergeyRoyt) [reply src/components/Badge/__stories__/Badge.stories.js:15]: Tried to do it once btw and found smth strange about it...
Not sure in which story it was (maybe in Dropdown or smth) and which actions it was, but eventually I manually added a few new actions to the array and it somehow broke functionality of the component itself... I was very surprised and gave up this idea for a while...

File: src/components/Menu/MenuItem/__stories__/MenuItem.stories.js (hunk 16 above)
Comment (SergeyRoyt) [reply src/components/Menu/MenuItem/__stories__/MenuItem.stories.js:15]: `iconPropsNameArray` is giving  you ability to pick icon from dropdown list for specific props
image


Maybe we can think of other way to do it. Anyway I think it's a bit out of scope of this one (this one is already enormous), but feel free to add it to the backlog!

File: src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx (hunk 15 above)
Comment (talkor) [reply src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx:129]: Oh no

File: src/components/Accordion/Accordion/__tests__/accordion.interactions.ts (hunk 4 above)
Comment (SergeyRoyt) [reply src/components/Accordion/Accordion/__tests__/accordion.interactions.ts:31]: That's what you've meant @shlomitc ?

File: src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx (hunk 15 above)
Comment (SergeyRoyt) [reply src/hooks/useKeyEvent/__stories__/useKeyEvent.mdx:129]: Not all `withoutSpacing`, don't worry, all good :) 
But lot's of them :D

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply package.json:159]: maybe use another package for it

File: CHANGELOG.md [hunk 21]
```diff
@@ -3183,7 +3183,7 @@
 
 #### Documentation
 * [#439](https://github.com/mondaycom/monday-ui-react-core/pull/439) Fixed code example of 'Menu with 2-depth sub menu' ([@laviomri](https://github.com/laviomri))
-* [#438](https://github.com/mondaycom/monday-ui-react-core/pull/438) fix links story controls, add leanylabs as contributors and add few description to props ([@hadasfa](https://github.com/hadasfa))
+* [#438](https://github.com/mondaycom/monday-ui-react-core/pull/438) fix links story add leanylabs as contributors and add few description to props ([@hadasfa](https://github.com/hadasfa))
```
Comment (YossiSaadi) [inline CHANGELOG.md:3186]: by mistake? 👀

File: babel.config.js [hunk 22]
```diff
@@ -53,6 +53,7 @@ module.exports = api => {
               }
         }
       ],
+      "@babel/preset-typescript",
```
Comment (YossiSaadi) [inline babel.config.js:56]: why do we need it now?

File: package.json [hunk 23]
```diff
@@ -72,8 +72,8 @@
     "build:esm": "rollup -c",
     "build:esm:mock-classnames": "mock_classnames=on rollup -c",
     "build:react-icons": "svg2react-icon --typescript --keep-colors node_modules/monday-ui-style/src/Icons src/components/Icon/Icons",
-    "storybook": "storybook=true start-storybook -p 7008",
-    "build-storybook": "npm run build:storybook-exports && storybook=true build-storybook",
+    "storybook": "storybook=true storybook dev -p 7008",
```
Comment (YossiSaadi) [inline package.json:75]: I saw you removed references that check 'storybook' env var in the setup of storybook (where you check IS_DEV), so is there's still need for storybook=true env variable?

File: package.json [hunk 24]
```diff
@@ -72,8 +72,8 @@
...
+    "build-storybook": "npm run build:storybook-exports && storybook=true storybook build",
```
Comment (YossiSaadi) [inline package.json:76]: same for above, storybook=true

File: package.json [hunk 25]
```diff
@@ -122,17 +122,20 @@
...
     "style-inject": "^0.3.0",
-    "vibe-storybook-components": "^0.14.0"
+    "vibe-storybook-components": "^0.14.2"
   },
   "devDependencies": {
     "@babel/core": "^7.23.2",
     "@babel/eslint-parser": "^7.16.5",
     "@babel/plugin-proposal-class-properties": "^7.16.5",
+    "@babel/plugin-proposal-private-methods": "^7.18.6",
+    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```
Comment (YossiSaadi) [inline package.json:132]: why are those needed?

File: src/utils/colors-vars-map.ts [hunk 26]
```diff
@@ -168,7 +168,7 @@ export const contentColors = [
   "pecan"
 ] as const;
 
-export type CONTENT_COLORS_VALUES = typeof contentColors[number];
+export type CONTENT_COLORS_VALUES = (typeof contentColors)[number];
```
[the remark this answers is not in the document]
Comment (SergeyRoyt) [reply src/utils/colors-vars-map.ts:171]: it became a lint issue after some dependency change - it doesn't, I believe

File: CHANGELOG.md (hunk 21 above)
Comment (SergeyRoyt) [reply CHANGELOG.md]: yep, removed by now
